ajax/add-ftp-user.php

```diff
@@ -0,0 +1,37 @@
+<?php
+require('../template/top.php');
+require('../api/groupme-funcs.php');
+
+if (isset($_POST)) {
+	$username = @$_POST['username'];
+	$password = @$_POST['password'];
+
+	do {
+		$response = json_decode(file_get_contents('https://www.google.com/recaptcha/api/siteverify?secret=' . GOOGLE_RECAPTCHA_KEY . '&response='.$captcha."&remoteip=".$_SERVER['REMOTE_ADDR']), true);
+        if (strlen($username) < 4) {
+			echo 'INVALID_NAME';
+			break;
+		} else if (strlen($password) < 4) {
+          	echo 'INVALID_PASSWORD';
+          	break;
+        }
+
+
+		$q = $db->query('INSERT INTO ftpusers (name, passwd)
```

Comment:
This is good, although there is one more thing that we need to do: ensure there aren't duplicate users. Ideally, we should enforce this with a UNIQUE constraint on the database column itself, rather than in the code (although, it definitely doesn't hurt to check in the code too because you can handle the error much more nicely). Also need to have a discussion about including DB updates in PRs. I have an idea for how I want to do it.

Reply:
Can you add this to the live site's db?